pkg/api/resource/quantity_test.go

```diff
@@ -284,13 +292,19 @@ func TestQuantityString(t *testing.T) {
 	}
 	desired := &inf.Dec{} // Avoid modifying the values in the table.
 	for _, item := range table {
-		if item.in.Amount.Cmp(decZero) == 0 {
```

Comment:
I am fine reverting this part of PR as it was not the core issue.  Apologies for missing it, but it did test what the generated code produced